spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/launcher/TaskLauncherSink.java

```diff
@@ -55,15 +60,9 @@ public void taskLauncherSink(TaskLaunchRequest request) {
 	private void launchTask(TaskLaunchRequest taskLaunchRequest) {
 		Assert.notNull(taskLauncher, "TaskLauncher has not been initialized");
 		logger.info("Launching Task for the following resource " + taskLaunchRequest);
-		MavenResource resource = new MavenResource.Builder()
-				.artifactId(taskLaunchRequest.getArtifact())
-				.groupId(taskLaunchRequest.getTaskGroupId())
-				.version(taskLaunchRequest.getTaskVersion())
-				.extension(taskLaunchRequest.getTaskExtension())
-				.classifier(taskLaunchRequest.getTaskClassifier())
-				.build();
-		AppDefinition definition = new AppDefinition(taskLaunchRequest.getArtifact(), taskLaunchRequest.getProperties());
-		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource);
+		Resource resource = delegatingResourceLoader.getResource(taskLaunchRequest.getUri());
+		AppDefinition definition = new AppDefinition("Task-" + taskLaunchRequest.hashCode(), taskLaunchRequest.getProperties());
```

Comment:
I think we can do better for a name than just `Task-<SOME_MEANINGLESS_STRING>`...